### PR TITLE
Added Modifier.toReplaceWith

### PR DESCRIPTION
**Pull request by Dr15Jones**
Added the method toReplaceWith to Modifier. This function is used to replicate the state of the fromObj into the existing toObj. This allows a larger conversion of the toObj than can be handled to the toModify method.

### CODE REVIEW

**Comment by Dr15Jones** [comment]
please test

**Comment by Dr15Jones** [comment]
+1

File: FWCore/ParameterSet/python/Config.py [hunk 1]
```diff
@@ -1165,6 +1165,25 @@ def toModify(self,obj, func=None,**kw):
...
+        for p in toObj.parameterNames_():
+            delattr(toObj,p)
+        for p in fromObj.parameterNames_():
+            setattr(toObj,p,getattr(fromObj,p))
+        if isinstance(fromObj,_TypedParameterizable):
+            toObj._TypedParameterizable__type = fromObj._TypedParameterizable__type
+            
+    else:
+        raise TypeError("toReplaceWith does not work with type "+str(type(toObj)))
+
```
Comment (makortel) [inline FWCore/ParameterSet/python/Config.py]: @Dr15Jones Should there be a check for `isChosen()` somewhere in above?

Comment (Dr15Jones) [inline FWCore/ParameterSet/python/Config.py]: I think you are right. I should have written a test showing it doesn't apply if not chosen.

Comment (Dr15Jones) [inline FWCore/ParameterSet/python/Config.py]: #13220

**Pull request merged**